aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -113,21 +110,21 @@ NestedTensorImpl::NestedTensorImpl(
     at::Tensor buffer,
     at::Tensor nested_size_tensor,
     at::Tensor nested_stride_tensor,
-    const std::vector<int64_t>& offsets)
+    std::vector<int64_t> offsets)
     : TensorImpl(
+          Storage(buffer.storage()),
           generate_nested_key_set(buffer),
-          buffer.dtype(),
-          buffer.device()),
-      buffer_(std::move(buffer)),
+          buffer.dtype()),
+      buffer_size_(buffer.unsafeGetTensorImpl()->sizes().vec()),
       nested_size_tensor_(std::move(nested_size_tensor)),
       nested_stride_tensor_(std::move(nested_stride_tensor)),
-      offsets_(offsets),
+      offsets_(std::move(offsets)),
```

Comment:
any side effects from taking ownership over `offsets` now? doubtful but making sure

Reply:
I am 99.9% sure there isn't, this was a clangD suggestion honestly and I think the std::move() just means take ownership when you can. And for the majority of cases it will otherwies it will just ref_count bump like ususal

Reply:
I think it used to be that you were taking offsets by reference and so you had to copy it into offsets_. Now you take offsets by copy so it is ok to move it into offsets_.